src/services/testrail/testrail.service.ts

```diff
@@ -132,33 +130,24 @@ export default class TestrailService {
     return this.testRailClient.addResultsForCases(runId, testRailResults);
   }
 
-  async attachTestRunsToMilestone(runs: ITestRun[], milestone_id: number): Promise<ITestRun[]> {
-    return new Promise((resolve, reject) => {
-      if (!runs?.length) resolve([]);
-
-      const promises = runs.map((run) => {
-        const { suite_id, name, description, assignedto_id, include_all } = run;
-
-        // @ts-ignore because the type for payload is incorrect
-        return this.testRailClient.updateRun(run.id, {
-          ...run,
-          suite_id,
-          name,
-          description,
-          assignedto_id,
-          include_all,
-          milestone_id,
-        });
+  async attachTestRunsToMilestone(
+    runs: ITestRun[],
+    milestone_id: number,
+    case_ids: number[]
+  ): Promise<ITestRun[]> {
+    if (!runs.length) return Promise.resolve([]);
+
+    const promises = runs.map((run) => {
+      return this.testRailClient.updateRun(run.id, {
+        ...run,
+        milestone_id,
+        case_ids,
```

Comment:
Introducing case_ids is the only change in functionality. It is flagged as required by the type system.

Reply:
the type system is mostly wrong. Validate if this is required using testrail API documentation:
https://support.testrail.com/hc/en-us/articles/7077874763156-Runs